### PR TITLE
Reduce initial delay of Stargate readiness probe

### LINKED ISSUE #654: Reduce initial delay of Stargate readiness probe
**Is your feature request related to a problem? Please describe.**
The Stargate deployment readiness probe has an initial delay of 4 minutes. This means that the kubelet will wait 4 minutes before checking that the pods are ready. This really slows down startup down. It also means that clients from outside the cluster will not be able to access Stargate during this time.

**Describe the solution you'd like**
The Stargate deployment already uses an initContainer to wait for at least one Cassandra pod to become ready. It needs to be updated to wait until the whole C* cluster is ready (see #612 for details). Given the delay from the initContainer we should be able to safely reduce the initial delay substantially, maybe to 10 or 15 seconds. When #612 is fixed, we should be able to further reduce it. 

We are currently **not** setting the failure threshold. We can set that in conjunction with the reduced initial delay.

The initial delay of the liveness probe is also set to 4 minutes. That can probably be safely reduced quite a bit as well, but I don't think it will impact or delay when the pods become ready.

As part of the work for this it would be nice to see some analysis on what the impact is for using different different settings for the readiness probe as well as the liveness probe.

**Describe alternatives you've considered**

**Additional context**
I want to point out that when we implement controllers for k8ssandra and Stargate, we won't need the initContainer any more. The controller can periodically check on the CassandraDatacenter. When it is ready, it can then deploy Stargate. We already do this in reaper-operator for Reaper.

### PR DESCRIPTION
**Pull request by adutra**
**What this PR does**:

* Modifies Stargate init container to wait not for one, but for all nodes in datacenter 0.
* Modifies the initial delay for Stargate probes

**Which issue(s) this PR fixes**:

Fixes #654

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: charts/k8ssandra/templates/stargate/stargate.yaml
Comment (jsanda) [inline charts/k8ssandra/templates/stargate/stargate.yaml:114]: These sound good for defaults, but we should make the probes configurable.

**Review by jsanda — CHANGES_REQUESTED** [review]

File: CHANGELOG-1.2.md
Comment (emerkle826) [inline CHANGELOG-1.2.md:16]: nit: This should be moved down and grouped with the other `[ENHANCEMENT]` updates in the list.

File: charts/k8ssandra/templates/stargate/stargate.yaml
Comment (adutra) [reply charts/k8ssandra/templates/stargate/stargate.yaml]: Good idea, implemented in eac44a5.

File: charts/k8ssandra/templates/stargate/stargate.yaml
Comment (jsanda) [inline charts/k8ssandra/templates/stargate/stargate.yaml:114]: Since you also define the default in `values.yaml` the `default` function is unnecessary here. `values.yaml` is part of the chart just like template files. If someone or another charts uses the k8ssandra charts and supplies their own values either by way of command line args with `--set` and/or with a chart properties file, they will be merged with `k8ssandra/values.yaml`. This means that `livenessInitialDelaySeconds` and `readinessInitialDelaySeconds` will not be unset for example by using a new chart properties file.

Does it make sense to make the other probe properties configurable as well? I think it is a question worth asking, but I haven't done enough research to see what is done elsewhere to formulate an opinion. 

I will say this though. It would be nice if we could expose probes in a consistent way across components, although that could be difficult due to differences within components.

**Review by jsanda — CHANGES_REQUESTED** [review]

File: charts/k8ssandra/templates/stargate/stargate.yaml
Comment (adutra) [reply charts/k8ssandra/templates/stargate/stargate.yaml:114]: > Since you also define the default in values.yaml the default function is unnecessary here. values.yaml is part of the chart just like template files.

Gotcha, will change that, thanks!

> Does it make sense to make the other probe properties configurable as well?

We'd need to weigh the pros (increased configurability) and cons (increased configuration complexity). My personal position would be to wait until more configurability is actually requested by users.

Comment (jsanda) [reply charts/k8ssandra/templates/stargate/stargate.yaml:114]: >We'd need to weigh the pros (increased configurability) and cons (increased configuration complexity). My personal position would be to wait until more configurability is actually requested by users.

👍

**Review by jsanda — APPROVED** [review]

**Comment by adutra** [comment]
Rebased on main and squashed some commits. Waiting for CI before merging.

**Pull request merged**
